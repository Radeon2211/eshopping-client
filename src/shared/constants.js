import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
