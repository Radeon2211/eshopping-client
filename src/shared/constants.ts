import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
