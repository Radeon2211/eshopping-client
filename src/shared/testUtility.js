// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
