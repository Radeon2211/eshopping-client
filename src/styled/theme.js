const theme = {
  colors: {
    light1: '#FAFAFA',
    light1Transparent: '#FAFAFA88',
    light2: '#DADFE1',
    light3: '#BFBFBF',
    light4: '#666',
    blue: '#1E8BC3',
    blueLight: '#22A7F0',
    blueDark: '#67809F',
    green: '#00A790',
    greenDark: '#009680',
    red: '#F22613',
  },
  shadows: {
    level1: '0 0 2px rgba(0, 0, 0, .1)',
    level2: '0 0 2px rgba(0, 0, 0, .3)',
    level3: '0 0 3px rgba(0, 0, 0, .3)',
    level4: '0 0 5px rgba(0, 0, 0, .3)',
    level5: '0 0 8px rgba(0, 0, 0, .3)',
  },
  fonts: {
    logo: 'Lobster, cursive',
    heading: 'Ubuntu, sans-serif',
    text: 'Lato, sans-serif',
  },
  spacings: {
    level1: '0.8rem',
    level2: '1.2rem',
    level3: '2.4rem',
    level4: '3.6rem',
    level5: '4.8rem',
  },
  zIndexes: {
    level1: 10,
    level2: 100,
    level3: 1000,
    level4: 10000,
  },
};

export default theme;
