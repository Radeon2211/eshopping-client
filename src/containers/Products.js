import React, { useEffect, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from '../store/actions/indexActions';
import Heading from '../components/UI/Heading/Heading';
import ProductsAndFilters from '../components/ProductsAndFilters/ProductsAndFilters';
import { pages } from '../shared/constants';
import { getParamsWithoutPollution } from '../shared/utility';

const Products = (props) => {
  const {
    location: { search },
  } = props;

  const productsPerPage = useSelector((state) => state.ui.productsPerPage);
  const userProfile = useSelector((state) => state.auth.profile);

  const dispatch = useDispatch();
  const onFetchProducts = useCallback(
    (queryParams, pageType) => dispatch(actions.fetchProducts(queryParams, pageType)),
    [dispatch],
  );

  useEffect(() => {
    onFetchProducts(search, pages.ALL_PRODUCTS);
  }, [search, onFetchProducts, productsPerPage, userProfile]);

  const correctQueryParams = getParamsWithoutPollution(search);
  const { name } = correctQueryParams;
  let headingText = 'All products';
  if (name) {
    headingText = `Results for "${name}"`;
  }

  return (
    <>
      <Heading variant="h3">{headingText}</Heading>
      <ProductsAndFilters page={pages.ALL_PRODUCTS} />
    </>
  );
};

export default Products;
