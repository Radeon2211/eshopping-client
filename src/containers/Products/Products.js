import React, { useEffect, useCallback } from 'react';
import queryString from 'query-string';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from '../../store/actions/indexActions';
import Heading from '../../components/UI/Heading/Heading';
import ProductsAndFilters from '../../components/ProductsAndFilters/ProductsAndFilters';
import { pages } from '../../shared/constants';

const Products = (props) => {
  const {
    location: { search },
  } = props;

  const maxQuantityPerPage = useSelector((state) => state.ui.maxQuantityPerPage);
  const userProfile = useSelector((state) => state.auth.profile);

  const dispatch = useDispatch();
  const onFetchProducts = useCallback(
    (queryParams, pageType) => dispatch(actions.fetchProducts(queryParams, pageType)),
    [dispatch],
  );

  useEffect(() => {
    onFetchProducts(search, pages.ALL_PRODUCTS);
  }, [search, onFetchProducts, maxQuantityPerPage, userProfile]);

  const { name } = queryString.parse(search);
  let headingText = 'All products';
  if (name) {
    headingText = `Results for "${name}"`;
  }

  return (
    <>
      <Heading variant="h3">{headingText}</Heading>
      <ProductsAndFilters page={pages.ALL_PRODUCTS} />
    </>
  );
};

export default Products;
