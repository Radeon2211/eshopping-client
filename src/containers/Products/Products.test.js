import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import matchMediaPolyfill from 'mq-polyfill';
import { Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Products from './Products';
import theme from '../../styled/theme';
import { defaultUserProfile, createProductItem } from '../../shared/testUtility/testUtility';
import { PRODUCTS_PER_PAGE } from '../../shared/constants';

const mockStore = configureMockStore([thunk]);

const defaultProducts = [
  createProductItem({
    id: 'p1',
    sellerUsername: 'user1',
    price: 10.6,
    quantity: 4,
    name: 'product1',
  }),
  createProductItem({
    id: 'p2',
    sellerUsername: 'user1',
    price: 299.98,
    quantity: 6,
    name: 'product2',
  }),
];

const setUp = (search = '?p=1') => {
  const props = {
    location: { pathname: '/products', search },
  };

  const history = {
    listen: jest.fn(),
    createHref: jest.fn(),
    location: { pathname: '/products', search },
  };

  const store = mockStore({
    auth: {
      profile: defaultUserProfile,
    },
    product: {
      products: defaultProducts,
      productCount: defaultProducts.length,
      minPrice: 10.6,
      maxPrice: 299.98,
    },
    ui: {
      productsPerPage: PRODUCTS_PER_PAGE,
      isDataLoading: false,
    },
  });

  return render(
    <Router history={history}>
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Products {...props} />
        </ThemeProvider>
      </Provider>
    </Router>,
  );
};

beforeAll(() => {
  matchMediaPolyfill(window);
  window.resizeTo = function resizeTo(width, height) {
    Object.assign(this, {
      innerWidth: width,
      innerHeight: height,
    }).dispatchEvent(new this.Event('resize'));
  };
});

afterEach(cleanup);

describe('<Products />', () => {
  describe('Check how renders', () => {
    it('should render everything correctly with default params and two products', () => {
      window.resizeTo(1920, 1080);
      const { asFragment } = setUp();
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render heading with `Results for "mushrooms"`', () => {
      setUp('?p=1&name=mushrooms');
      expect(screen.getByTestId('Products-heading')).toHaveTextContent('Results for "mushrooms"');
    });

    it('should render heading with `Results for "wellingtons"` if wellingtons is in last occurrence of name param', () => {
      setUp('?p=1&name=mushrooms&name=wellingtons');
      expect(screen.getByTestId('Products-heading')).toHaveTextContent('Results for "wellingtons"');
    });
  });
});
