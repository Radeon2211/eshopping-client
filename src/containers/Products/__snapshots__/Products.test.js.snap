// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> Check how renders should render everything correctly with default params and two products 1`] = `
<DocumentFragment>
  <h3
    class="sc-bdfBwQ hSpIvR"
    data-testid="Products-heading"
  >
    All products
  </h3>
  <div
    class="sc-gsTCUz iakHmb"
  >
    <section
      class="sc-gKsewC hOOrhu"
      data-testid="Filters"
    >
      <div
        class="sc-dlfnbm hnHwOk"
      >
        <div
          class="select css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Default sorting
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-3-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="sortBy"
            type="hidden"
            value=""
          />
        </div>
        <div
          class="sc-iqHYGH cgHQme"
        >
          <div
            class="sc-crrsfI vUjlw"
          >
            <span
              class="sc-iBPRYJ kYvbgz"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Price ($)
            </span>
            <div
              class="inputs"
            >
              <input
                class="sc-bqyKva frbvYy"
                data-testid="NumberInput-minPrice"
                name="minPrice"
                type="number"
                value="10.6"
              />
              <span
                class="inputs-gap"
              >
                â€”
              </span>
              <input
                class="sc-bqyKva frbvYy"
                data-testid="NumberInput-maxPrice"
                name="maxPrice"
                type="number"
                value="299.98"
              />
            </div>
          </div>
          <div
            class="sc-dQppl eA-dttf"
          >
            <input
              class="input min"
              data-testid="PriceSlider-price-range-min"
              max="299.98"
              min="10.6"
              name="minPrice"
              step="0.29998"
              type="range"
              value="10.6"
            />
            <input
              class="input max"
              data-testid="PriceSlider-price-range-max"
              max="299.98"
              min="10.6"
              name="maxPrice"
              step="0.29998"
              type="range"
              value="299.98"
            />
            <div
              class="track"
            />
            <div
              class="range"
            />
            <div
              class="thumb left"
            />
            <div
              class="thumb right"
            />
          </div>
        </div>
        <div
          class="sc-eCssSg fEdOcl"
        >
          <span
            class="sc-iBPRYJ kYvbgz"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            Condition
          </span>
          <div
            class="sc-jSgupP gZkkyh"
          >
            <input
              data-testid="Filters-checkbox-new"
              id="new"
              name="new"
              type="checkbox"
            />
            <label
              for="new"
            >
              new
            </label>
          </div>
          <div
            class="sc-jSgupP gZkkyh"
          >
            <input
              data-testid="Filters-checkbox-used"
              id="used"
              name="used"
              type="checkbox"
            />
            <label
              for="used"
            >
              used
            </label>
          </div>
          <div
            class="sc-jSgupP gZkkyh"
          >
            <input
              data-testid="Filters-checkbox-not-applicable"
              id="not_applicable"
              name="not_applicable"
              type="checkbox"
            />
            <label
              for="not_applicable"
            >
              not applicable
            </label>
          </div>
        </div>
        <button
          class="sc-fubCfw dqKGwi"
          color="blue"
          data-testid="Filters-submit-btn"
          type="button"
        >
          Filter
        </button>
      </div>
    </section>
    <section
      class="sc-gKsewC hOOrhu"
      data-testid="ProductsAndFilters-product-list-section"
    >
      <div
        class="sc-fKFyDc gMvzTj"
      >
        <div
          class="sc-dmlrTW fHfiAZ"
        >
          <label
            class="label"
            for="perPageQuantity"
          >
            Per page
          </label>
          <div
            class="select css-2b097c-container"
            id="perPageQuantity"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-l640om-control"
            >
              <div
                class=" css-3cb03z-ValueContainer"
              >
                <div
                  class=" css-rrw4q2-SingleValue"
                >
                  10
                </div>
                <input
                  aria-autocomplete="list"
                  class="css-wmatm6-dummyInput-DummyInput"
                  id="react-select-2-input"
                  readonly=""
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-ml9fjp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-hHftDr fbtuLP"
          data-testid="InputPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-left-arrow"
          >
            <div
              class="sc-kstrdz lgDWlP"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <form
            class="form-number"
          >
            <input
              class="sc-bqyKva jja-dMM"
              data-testid="NumberInput-page"
              name="page"
              type="number"
              value="1"
            />
          </form>
          <span
            class="sc-iBPRYJ jlwgZf"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            of
          </span>
          <span
            class="sc-iBPRYJ jlwgZf"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-right-arrow"
          >
            <div
              class="sc-kstrdz dfYKpJ"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="sc-bkzZxe fwDVgA"
        data-testid="ProductList"
      >
        <a
          data-testid="ProductItem"
        >
          <div
            class="sc-hBEYos jltyQI sc-fodVxV iia-DsQ"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <div
              class="sc-hBEYos ezBczf sc-fFubgz kUYmUr"
              data-testid="ProductThumbnail"
              direction="row"
              height="15"
              spacing=""
              width="13"
              wrap="nowrap"
            >
              <img
                alt="product1"
                data-testid="ProductThumbnail-img"
                src="no-photo.png"
              />
            </div>
            <div
              class="sc-hBEYos fRCKMd"
              direction="column"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-iBPRYJ kYvbgz"
                color=""
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="0px"
                transform=""
              >
                product1
              </span>
              <span
                class="sc-iBPRYJ BBDCp"
                color=""
                data-testid="ProductItem-condition"
                display="inline-block"
                overflow="visible"
                size="1"
                spacing="0px"
                transform=""
              >
                <span
                  class="sc-iBPRYJ NtwOd"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  spacing="0px"
                  transform=""
                >
                  Condition:
                </span>
                 New
              </span>
              <span
                class="sc-iBPRYJ jtiZdA"
                color=""
                data-testid="ProductItem-price"
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $10.60
              </span>
            </div>
          </div>
        </a>
        <a
          data-testid="ProductItem"
        >
          <div
            class="sc-hBEYos jltyQI sc-fodVxV iia-DsQ"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <div
              class="sc-hBEYos ezBczf sc-fFubgz kUYmUr"
              data-testid="ProductThumbnail"
              direction="row"
              height="15"
              spacing=""
              width="13"
              wrap="nowrap"
            >
              <img
                alt="product2"
                data-testid="ProductThumbnail-img"
                src="no-photo.png"
              />
            </div>
            <div
              class="sc-hBEYos fRCKMd"
              direction="column"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-iBPRYJ kYvbgz"
                color=""
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="0px"
                transform=""
              >
                product2
              </span>
              <span
                class="sc-iBPRYJ BBDCp"
                color=""
                data-testid="ProductItem-condition"
                display="inline-block"
                overflow="visible"
                size="1"
                spacing="0px"
                transform=""
              >
                <span
                  class="sc-iBPRYJ NtwOd"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  spacing="0px"
                  transform=""
                >
                  Condition:
                </span>
                 New
              </span>
              <span
                class="sc-iBPRYJ jtiZdA"
                color=""
                data-testid="ProductItem-price"
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $299.98
              </span>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-dIUggk gVtHox"
      >
        <span
          class="sc-iBPRYJ inTSQh"
          color="#777"
          display="inline-block"
          overflow="visible"
          size="1"
          spacing="0px"
          transform=""
        >
          1 - 2 of 2 products
        </span>
        <div
          class="sc-idOhPF gwyQBe"
          data-testid="NumberPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-left-arrow"
          >
            <div
              class="sc-kstrdz lgDWlP"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <a
            class="number-link active"
            data-testid="NumberPagination-page1"
          >
            1
          </a>
          <span
            class="of"
          >
            of
          </span>
          <span
            class="of"
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-right-arrow"
          >
            <div
              class="sc-kstrdz dfYKpJ"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
